pkg/storage/stores/indexshipper/compactor/compactor_client.go

```diff
@@ -0,0 +1,42 @@
+package compactor
+
+import (
+	"flag"
+	"net/http"
+	"time"
+
+	"github.com/grafana/dskit/crypto/tls"
+)
+
+// Config for compactor's generation-number client
+type ClientConfig struct {
+	TLSEnabled bool             `yaml:"tls_enabled"`
+	TLS        tls.ClientConfig `yaml:",inline"`
+}
+
+// RegisterFlags adds the flags required to config this to the given FlagSet.
+func (cfg *ClientConfig) RegisterFlags(f *flag.FlagSet) {
+	prefix := "boltdb.shipper.compactor.client"
+	f.BoolVar(&cfg.TLSEnabled, prefix+".tls-enabled", false,
```

Comment:
Is the separate `tls-enabled` configuration necessary? If I remember correctly, then the configuration contains the whole URL to the compactor including the scheme, so that should be enough for the client to decide whether to use TLS or not.